src/code/z_play.c

```diff
@@ -260,53 +255,50 @@ void Gameplay_Init(GameState* thisx) {
     }
 
-    if ((gSaveContext.dayTime >= 0xC001) || (gSaveContext.dayTime < 0x4555)) {
+    if (gSaveContext.dayTime > 0xC000 || gSaveContext.dayTime < 0x4555) {
         gSaveContext.nightFlag = 1;
     } else {
         gSaveContext.nightFlag = 0;
     }
 
     Cutscene_HandleConditionalTriggers(globalCtx);
 
-    if ((gSaveContext.gameMode != 0) || (gSaveContext.cutsceneIndex >= 0xFFF0)) {
+    if (gSaveContext.gameMode != 0 || gSaveContext.cutsceneIndex >= 0xFFF0) {
         gSaveContext.nayrusLoveTimer = 0;
         func_800876C8(globalCtx);
         gSaveContext.sceneSetupIndex = (gSaveContext.cutsceneIndex & 0xF) + 4;
-    } else if (LINK_IS_CHILD && (gSaveContext.nightFlag == 0)) {
+    } else if (LINK_IS_CHILD && gSaveContext.nightFlag == 0) {
         gSaveContext.sceneSetupIndex = 0;
-    } else if (LINK_IS_CHILD && (gSaveContext.nightFlag != 0)) {
+    } else if (LINK_IS_CHILD && gSaveContext.nightFlag != 0) {
         gSaveContext.sceneSetupIndex = 1;
-    } else if (LINK_IS_ADULT && (gSaveContext.nightFlag == 0)) {
+    } else if (LINK_IS_ADULT && gSaveContext.nightFlag == 0) {
         gSaveContext.sceneSetupIndex = 2;
     } else {
         gSaveContext.sceneSetupIndex = 3;
     }
 
     tempSetupIndex = gSaveContext.sceneSetupIndex;
-    if ((gEntranceTable[gSaveContext.entranceIndex].scene == SCENE_SPOT00) && LINK_IS_CHILD &&
-        (gSaveContext.sceneSetupIndex < 4)) {
+    if ((gEntranceTable[((void)0, gSaveContext.entranceIndex)].scene == SCENE_SPOT00) && LINK_IS_CHILD &&
+        gSaveContext.sceneSetupIndex < 4) {
         if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD) && CHECK_QUEST_ITEM(QUEST_GORON_RUBY) &&
             CHECK_QUEST_ITEM(QUEST_ZORA_SAPPHIRE)) {
             gSaveContext.sceneSetupIndex = 1;
         } else {
             gSaveContext.sceneSetupIndex = 0;
         }
-    } else if ((gEntranceTable[gSaveContext.entranceIndex].scene == SCENE_SPOT04) && LINK_IS_ADULT &&
-               (gSaveContext.sceneSetupIndex < 4)) {
-        if (gSaveContext.eventChkInf[4] & 0x100) {
-            gSaveContext.sceneSetupIndex = 3;
-        } else {
-            gSaveContext.sceneSetupIndex = 2;
-        }
+    } else if ((gEntranceTable[((void)0, gSaveContext.entranceIndex)].scene == SCENE_SPOT04) && LINK_IS_ADULT &&
+               gSaveContext.sceneSetupIndex < 4) {
+        gSaveContext.sceneSetupIndex = (gSaveContext.eventChkInf[4] & 0x100) ? 3 : 2;
     }
 
-    spawnEntrance = &gEntranceTable[gSaveContext.entranceIndex + gSaveContext.sceneSetupIndex];
-    Gameplay_SpawnScene(globalCtx, spawnEntrance->scene, spawnEntrance->spawn);
-    osSyncPrintf("\nSCENE_NO=%d COUNTER=%d\n", gSaveContext.entranceIndex, gSaveContext.sceneSetupIndex);
+    Gameplay_SpawnScene(
+        globalCtx, (&gEntranceTable[(((void)0, gSaveContext.entranceIndex)) + (((void)0, gSaveContext.sceneSetupIndex))])->scene,
+        (&gEntranceTable[(((void)0, gSaveContext.sceneSetupIndex)) + (((void)0, gSaveContext.entranceIndex))])->spawn);
```

Comment:
Any chance these lines can be made a bit cleaner like this?
```suggestion
        globalCtx, gEntranceTable[((void)0, gSaveContext.entranceIndex) + ((void)0, gSaveContext.sceneSetupIndex)].scene,
        gEntranceTable[((void)0, gSaveContext.sceneSetupIndex) + ((void)0, gSaveContext.entranceIndex)].spawn);
```